flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/GroupAggProcessFunction.scala

```diff
@@ -118,19 +131,33 @@ class GroupAggProcessFunction(
       }
     }
 
-    state.update(accumulators)
+    if (inputCnt > 0) {
```

Comment:
thanks for fixing this bug. I think we also need to delete `override def consumesRetractions = true` in `DataStreamGroupAggregate`. Because, this fix will make `DataStreamGroupAggregate` forwarding retraction messages. Also tests in `RetractionRulesTest` need to be updated